### ISSUE TITLE
[Bug]: Loading more than one layer with a 'forward' validity range causes the app to hang

### ISSUE THREAD

**Issue by wadhwamatic** [WFP-VAM/prism-app#789]
### What happened?

If I configure and use two layers with 'forward' validity date ranges, the app hangs

1. Use this branch: https://github.com/WFP-VAM/prism-app/tree/rbd-multi-dates. COUNTRY=rbd
2. Activate Cadre Harmonise > Phase classification > Phase classification
3. Activate Rainfall > Rainfall Amount > Rainfall Aggregate 10-day. The app will hang when the second layer is activated

 

### Which country / deployment are you running?

RBD on this branch: https://github.com/WFP-VAM/prism-app/tree/rbd-multi-dates

**Comment by HarryMytilinaios** [comment]
The error was due an infinite rendering of the `Mapview` component. Unfortunately this error persists. I tried to refactor the Mapview component a bit by splitting and decoupling the logic a little bit. I created a unique autonomous component which will be the `MapComponent` itself whith each layers. Also created the `useMapOnClick` as an independent hook to have a more cleaner and readable components and functions. I tried understanding the `serverAvailableDates`, reduced the available dates also to see if the problem was in the big array length, but the problem of the infinite rendering still persists, in my case I think it has to do something with the useEffects inside the `MapView` component. We 4 different `useEffect` hooks that run when the component initializes.
